Include/internal/pycore_global_dict_state.h

```diff
@@ -8,33 +8,14 @@ extern "C" {
 #  error "this header requires Py_BUILD_CORE define"
 #endif
 
-
-#ifndef WITH_FREELISTS
-// without freelists
-#  define PyDict_MAXFREELIST 0
-#endif
-
-#ifndef PyDict_MAXFREELIST
-#  define PyDict_MAXFREELIST 80
-#endif
-
 #define DICT_MAX_WATCHERS 8
 
-struct _Py_dict_state {
+struct _Py_global_dict_state {
```

Comment:
Maybe `_Py_dict_interp_state`?

In some places we use "global_state" to mean runtime-global (not interpreter). For example, ew have:
* _dtoa_interp_state (per-interpreter)
* _mimalloc_interp_state (per-interpreter)
* _obmalloc_global_state (in PyRuntime)